src/node.rs

```diff
@@ -0,0 +1,170 @@
+use crate::{Block, Consensus, Key, Message, Network, PeerId, State};
+
+pub trait Node {
+    type Block: Block<Payload = Self::Payload, PeerId = Self::PeerId>;
+    type Key: Key;
+    type Message: Message<Block = Self::Block, Key = Self::Key, PeerId = Self::PeerId>;
+    type Network: Network<Message = Self::Message, Payload = Self::Payload, PeerId = Self::PeerId>;
+    type Payload;
+    type PeerId: PeerId;
```

Comment:
Since peers could be non-validating nodes or light clients, we should go with something more like `AuthorityId` or `ValidatorId`